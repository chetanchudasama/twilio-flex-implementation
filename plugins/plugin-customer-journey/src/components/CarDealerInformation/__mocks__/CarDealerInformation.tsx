import React from "react";
import { Props } from "../CarDealerInformation";

const CarDealerInformation: React.FC<Props> = (props) => <div />;

export default CarDealerInformation;
