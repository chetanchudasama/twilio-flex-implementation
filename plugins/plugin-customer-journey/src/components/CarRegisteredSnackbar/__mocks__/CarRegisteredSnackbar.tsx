import React from "react";
import { CarRegisteredSnackbarProps } from "../CarRegisteredSnackbar.Props";

export const CarRegisteredSnackbar: React.FC<CarRegisteredSnackbarProps> = (
  props
) => {
  return <div className="car-registered-snackbar">CarRegisteredSnackbar</div>;
};
