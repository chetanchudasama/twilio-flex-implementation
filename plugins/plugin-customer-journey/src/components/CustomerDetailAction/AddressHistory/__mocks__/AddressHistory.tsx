import React from "react";
import { AddressHistoryProps } from "../AddressHistory";

const AddressHistory: React.FC<AddressHistoryProps> = (props) => (
  <div className="address-history" />
);

export default AddressHistory;
