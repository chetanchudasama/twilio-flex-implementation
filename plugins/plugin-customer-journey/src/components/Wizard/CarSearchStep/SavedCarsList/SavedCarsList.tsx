import React from "react";
import { Shared } from "@common/components";
import { Grid } from "@material-ui/core";
import { v4 } from "uuid";
import CarSearchResultDetailWrapper from "../CarSearchResultDetailWrapper/CarSearchResultDetailWrapper.Container";
import { VehicleSortingOptions } from "../../../../common/enum";
import { StateToProps, DispatchToProps } from "./SavedCarsList.Props";
import { CarSearchResultStyles } from "./SavedCarsList.Styles";
import { SortDropdown } from "../../SortDropdown/SortDropdown";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface OwnProps {}

export type ComponentProps = StateToProps & DispatchToProps & OwnProps;

const SavedCarsList: React.FC<ComponentProps> = ({
  sort,
  setSort,
  savedVehicles,
  totalSavedVehicles,
}) => {
  const onSortChange = (event: React.ChangeEvent<HTMLSelectElement>) =>
    setSort(
      Number(event.target.value) ?? VehicleSortingOptions.PriceDescending
    );

  const noSavedVehiclesComponent = (
    <Grid item xs={12} className="no-result">
      No cars saved.
    </Grid>
  );

  const savedVehiclesComponent = (
    <>
      <Grid item xs={12} className="result-sort-container">
        <Grid
          container
          direction="row"
          justify="space-between"
          alignItems="center"
        >
          <Grid item className="result-text">
            {`${Shared.getFormattedNumber(totalSavedVehicles)} ${
              totalSavedVehicles > 1 ? "results" : "result"
            }`}
          </Grid>
          <Grid item>
            <SortDropdown sort={sort} onSortChange={onSortChange} />
          </Grid>
        </Grid>
      </Grid>
      {savedVehicles.map((vehicle) => (
        <Grid item xs={12} key={v4()}>
          <CarSearchResultDetailWrapper vehicleDetail={vehicle} />
        </Grid>
      ))}
    </>
  );

  return (
    <CarSearchResultStyles>
      {savedVehicles.length ? savedVehiclesComponent : noSavedVehiclesComponent}
    </CarSearchResultStyles>
  );
};

export default SavedCarsList;
