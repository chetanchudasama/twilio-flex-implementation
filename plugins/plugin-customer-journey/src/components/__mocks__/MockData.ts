import { WorkerAttributes } from "@twilio/flex-ui";
import { namespace } from "../../states";

const workerAttributes: WorkerAttributes = {
  contact_uri: "client:test",
  email: "test@test.com",
  full_name: "test user",
  image_url: "img_url",
  roles: ["admin"],
  agentId: 1,
};

const conferenceSource = {
  source: {
    participants: [
      {
        callSid: "callSid",
        participantType: "customer",
        workerSid: "",
      },
      {
        callSid: "callSid",
        participantType: "worker",
        workerSid: "workerSid",
      },
    ],
  },
};

const conferenceState = new Map();
conferenceState.set("taskSid", conferenceSource);

export const flexManagerMock = jest.fn().mockReturnValue({
  workerClient: {
    source: {},
    sid: "workerSid",
    name: "test user",
    fullName: "test user",
    activityName: "Offline",
    attributes: workerAttributes,
    dateUpdated: new Date(),
    isAvailable: false,
    activityDuration: "",
    dateStatusChanged: new Date(1).toISOString(),
    activity: {
      available: false,
      name: "Offline",
      sid: "OfflineSid",
    },
  },
  store: {
    getState: jest.fn().mockReturnValue({
      flex: {
        worker: {
          attributes: workerAttributes,
        },
        session: {
          ssoTokenPayload: {
            token: "ssoToken",
          },
        },
        conferences: {
          states: conferenceState,
        },
      },
      [namespace]: {
        phoneNumberList: {
          phoneNumberList: [
            {
              friendlyName: "Sales",
              phoneNumber: "+1234567891",
              queueSid: "QUEUE_SID",
            },
          ],
        },
      },
    }),
  },
  strings: {
    DIALPADExternalTransferSearchInputLabel: "Search By Name or PhoneNumber",
    DIALPADExternalTransferHoverOver: "Add",
    DIALPADExternalTransferPopupHeader: "External Transfer",
  },
  serviceConfiguration: {
    outbound_call_flows: {
      default: {
        caller_id: "CALLER_ID",
      },
    },
  },
});
