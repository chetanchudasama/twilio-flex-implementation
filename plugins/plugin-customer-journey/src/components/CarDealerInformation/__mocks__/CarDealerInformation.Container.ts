import CarDealerInformation from "./CarDealerInformation";

export default CarDealerInformation;
