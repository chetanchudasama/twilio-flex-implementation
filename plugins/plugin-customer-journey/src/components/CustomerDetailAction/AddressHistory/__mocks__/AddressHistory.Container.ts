import AddressHistory from "./AddressHistory";

export default AddressHistory;
