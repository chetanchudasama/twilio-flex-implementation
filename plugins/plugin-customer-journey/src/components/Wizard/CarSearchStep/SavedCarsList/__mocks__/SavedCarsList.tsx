import React from "react";
import { ComponentProps } from "../SavedCarsList";

const SavedCarsList: React.FC<ComponentProps> = (props) => <div />;

export default SavedCarsList;
