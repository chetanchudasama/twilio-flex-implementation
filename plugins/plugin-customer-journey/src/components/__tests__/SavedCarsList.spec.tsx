import React from "react";
import { create } from "react-test-renderer";
import flushPromises from "flush-promises";
import { Manager } from "@twilio/flex-ui";
import { mount } from "enzyme";
import { VehicleSortingOptions } from "../../common/enum";
import { VehicleSummaryDetailModel } from "../../models/VehicleSummaryDetailModel";
import CarSearchResultDetailWrapperContainer from "../Wizard/CarSearchStep/CarSearchResultDetailWrapper/CarSearchResultDetailWrapper.Container";
import SavedCarsList, {
  ComponentProps,
} from "../Wizard/CarSearchStep/SavedCarsList/SavedCarsList";
import { flexManagerMock } from "../__mocks__/MockData";

jest.mock(
  "../Wizard/CarSearchStep/CarSearchResultDetailWrapper/CarSearchResultDetailWrapper.Container"
);
jest.mock(
  "../Wizard/CarSearchStep/CarSearchResultDetailWrapper/CarSearchResultDetailWrapper"
);

jest.mock("@common/components", () => {
  const components = jest.requireActual("@common/components");
  return {
    ...components,
  };
});

Manager.getInstance = flexManagerMock;

const defaultProps: ComponentProps = {
  savedVehicles: [],
  totalSavedVehicles: 0,
  sort: VehicleSortingOptions.PriceDescending,
  setSort: jest.fn(),
};

let props: ComponentProps = defaultProps;

Manager.getInstance = flexManagerMock;

describe("SavedCarsList", () => {
  beforeEach(() => {
    props = { ...defaultProps };
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.clearAllTimers();
  });

  it("renders correctly, if no vehicle found", async () => {
    const wrapper = create(<SavedCarsList {...props} />).toJSON();
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly, if vehicle found in search list", async () => {
    props.totalSavedVehicles = 1;
    props.savedVehicles = [new VehicleSummaryDetailModel()];

    const wrapper = create(<SavedCarsList {...props} />).toJSON();
    await flushPromises();
    expect(wrapper).toMatchSnapshot();
  });

  it("should render vehicle details wrapper component if vehicles present", async () => {
    props.totalSavedVehicles = 1;
    props.savedVehicles = [new VehicleSummaryDetailModel()];

    const wrapper = mount(<SavedCarsList {...props} />);
    await flushPromises();
    wrapper.update();

    expect(
      wrapper.find(CarSearchResultDetailWrapperContainer).exists()
    ).toBeTruthy();
  });

  it("should not render vehicle details wrapper component if vehicles are not present", async () => {
    props.totalSavedVehicles = 0;
    props.savedVehicles = [];

    const wrapper = mount(<SavedCarsList {...props} />);
    await flushPromises();
    wrapper.update();

    expect(
      wrapper.find(CarSearchResultDetailWrapperContainer).exists()
    ).toBeFalsy();
  });

  it("should show no cars found message if no vehicles found", async () => {
    props.totalSavedVehicles = 0;
    props.savedVehicles = [];

    const wrapper = mount(<SavedCarsList {...props} />);
    await flushPromises();
    wrapper.update();

    expect(wrapper.html()).toContain("No cars saved.");
  });

  it("should show number of vehicles found in the search", async () => {
    props.totalSavedVehicles = 1;
    props.savedVehicles = [new VehicleSummaryDetailModel()];

    const wrapper = mount(<SavedCarsList {...props} />);
    await flushPromises();
    wrapper.update();

    expect(wrapper.html()).toContain("1 result");
  });
});
