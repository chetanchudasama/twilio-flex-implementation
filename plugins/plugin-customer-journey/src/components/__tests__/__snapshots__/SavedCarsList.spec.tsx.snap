// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedCarsList renders correctly, if no vehicle found 1`] = `
<div
  className="css-1ma0a9h"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 no-result"
  >
    No cars saved.
  </div>
</div>
`;

exports[`SavedCarsList renders correctly, if vehicle found in search list 1`] = `
<div
  className="css-1ma0a9h"
>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41 result-sort-container"
  >
    <div
      className="MuiGrid-container-1 MuiGrid-align-items-xs-center-9 MuiGrid-justify-xs-space-between-20"
    >
      <div
        className="MuiGrid-item-2 result-text"
      >
        1 result
      </div>
      <div
        className="MuiGrid-item-2"
      >
        <div
          className="MuiFormControl-root-103 MuiFormControl-marginDense-105 MuiFormControl-fullWidth-106 CustomSelectContent-formControl-98"
        >
          <div
            className="MuiInputBase-root-127 MuiOutlinedInput-root-114 MuiInputBase-formControl-128 MuiInputBase-marginDense-134 CustomSelectContent-select-100 "
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="MuiPrivateNotchedOutline-root-144 MuiOutlinedInput-notchedOutline-121"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="MuiPrivateNotchedOutline-legend-145"
                style={
                  Object {
                    "width": 0,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <div
              className="MuiSelect-root-107"
            >
              <div
                aria-haspopup="true"
                aria-pressed="false"
                className="MuiSelect-select-108 MuiSelect-selectMenu-111 MuiInputBase-input-137 MuiOutlinedInput-input-122 MuiInputBase-inputMarginDense-138 MuiOutlinedInput-inputMarginDense-123"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                role="button"
                tabIndex={0}
              >
                Sort by: Price (High to Low)
              </div>
              <input
                type="hidden"
                value={2}
              />
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root-146 MuiSelect-icon-113"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
  >
    <div />
  </div>
</div>
`;
